nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
We should protect against having multiple instances of the validator. Only one instance is required. Maybe one way to do that is to override the `equals` and `hashCode` method for `RequiredFieldValidator` so that they return a fixed value regardless of the instance. Let me know if you have other ideas.